### ISSUE TITLE
nullptr access closing a CFloatingDockContainer

### ISSUE THREAD

**Issue by jankrassnigg** [githubuser0xFFFF/Qt-Advanced-Docking-System#171]
I sometimes get a crash when closing a floating dock container with one or multiple dock widgets tabbed in it. Not sure what exactly happens, but here is what I'm seeing.

1. I made one document floating, then added a couple of other documents to that floating dock area.
1. I clicked the X for the entire floater, closing all documents
1. It crashes in `CDockWidget::icon() `with `this == nullptr`

Callstack:

```cpp
>	ads.dll!ads::CDockWidget::icon() Line 634	C++
 	ads.dll!ads::FloatingDockContainerPrivate::reflectCurrentWidget(ads::CDockWidget * CurrentWidget) Line 131	C++
 	ads.dll!ads::CFloatingDockContainer::updateWindowTitle() Line 661	C++
 	ads.dll!ads::CDockWidget::toggleViewInternal(bool Open) Line 523	C++
 	ads.dll!ads::CDockWidget::toggleView(bool Open) Line 472	C++
 	ads.dll!ads::CFloatingDockContainer::hideEvent(QHideEvent * event) Line 462	C++
 	Qt5Widgetsd.dll!00007ffa8ccfa49f()	Unknown
 	ads.dll!ads::CFloatingDockContainer::event(QEvent * e) Line 545	C++
 	Qt5Widgetsd.dll!00007ffa8cc95552()	Unknown
 	Qt5Widgetsd.dll!00007ffa8cc92b83()	Unknown
```

Basically, `CFloatingDockContainer::updateWindowTitle()` calls `TopLevelDockArea->currentDockWidget()` which returns `nullptr` and feeds that into `d->reflectCurrentWidget(CurrentWidget);` which then leads to the crash.

Not sure whether `updateWindowTitle` should just check for nullptr or whether the fact that `currentDockWidget` can return nullptr at that time is already invalid. However, doing a nullptr check before calling `reflectCurrentWidget` seems to be enough to fix the issue.

**Comment by githubuser0xFFFF** [comment]
Thank you for this bug report. Is this also reproducible with the demo application? Could you please tell me the `CDockManager` configuration flags that you use?

**Comment by githubuser0xFFFF** [comment]
I could not reproduce the problem in the demo application with MSVC and GCC. I implemented the proposed solution to fix the potential problem.

**Comment by jankrassnigg** [comment]
CDockManager flags are:

ads::CDockManager::DockAreaHasCloseButton
ads::CDockManager::DockAreaCloseButtonClosesTab
ads::CDockManager::OpaqueSplitterResize
ads::CDockManager::AllTabsHaveCloseButton

Tested this in a Debug build, VS 2017 (15.9.23)

I'll have to see whether I find the time to test this in the demo application.
